### ISSUE TITLE
Test the model on the test dataset after freezing the weights.

### ISSUE THREAD

**Issue by alisterburt** [warpem/miss-alignment#7]

**Comment by McHaillet** [comment]
I added some code to evaluate the test dataset and generate attention maps for this data (still branch `test_loss_regu`). Scores are higher on high quality (aligned) map, but that is expected as this is still that same converged model that had the scores inversed. So, it correctly ranks aligned/misaligned in the test data as well.

```bash
model = emd_24189_10A
High quality map has a score of tensor([[-14.6817]])
Low quality map has a score of tensor([[-17.2197]])
model = emd_26916_10A
High quality map has a score of tensor([[-8.6292]])
Low quality map has a score of tensor([[-12.1833]])
model = emd_27962_10A
High quality map has a score of tensor([[-0.5542]])
Low quality map has a score of tensor([[-7.2635]])
model = emd_33653_10A
High quality map has a score of tensor([[-12.0610]])
Low quality map has a score of tensor([[-18.4057]])
model = emd_38726_10A
High quality map has a score of tensor([[3.0719]])
Low quality map has a score of tensor([[0.6598]])
model = emd_38909_10A
High quality map has a score of tensor([[-13.1760]])
Low quality map has a score of tensor([[-19.2603]])
model = emd_39246_10A
High quality map has a score of tensor([[18.5174]])
Low quality map has a score of tensor([[11.0540]])
model = emd_41006_10A
High quality map has a score of tensor([[-22.6860]])
Low quality map has a score of tensor([[-24.6723]])
model = emd_45293_10A
High quality map has a score of tensor([[12.5070]])
Low quality map has a score of tensor([[1.1576]])
model = emd_48372_10A
High quality map has a score of tensor([[1.4469]])
Low quality map has a score of tensor([[-3.2678]])
```

Here is an example of the attention map for emd_45293, top row is misaligned, bottom row is aligned:

https://github.com/user-attachments/assets/ddf432d7-1679-42f5-832f-459b067cd92e

**Comment by alisterburt** [comment]
Very nice work! Am I reading the "attention map" right that it's focusing mostly above the object? Are you seeing anything interesting in any of these maps?

**Comment by dtegunov** [comment]
Guys, let's try to prioritize this one because we may need to explore different losses / training strategies if it doesn't work – regardless of shift pattern realism.

**Comment by dtegunov** [comment]
One more thought here: If the loss landscape proves too bumpy, try generating multiple volumes with different content but identical misalignment, and then push them all through the model and average the gradients. This will be closer to what we'd be doing with full tomograms since they're much larger than 64^3.

You'll probably want something faster than plain GD or ADAM for this optimization, e.g. https://pytorch.org/docs/stable/generated/torch.optim.LBFGS.html

**Comment by dtegunov** [comment]
And another thought: since the correctness of the loss function is our biggest unknown right now, why not just disentangle the two problems and train using the misalignment std dev (which includes jitter and trajectories) for the loss calculation instead of the margin loss, then use that model to optimize alignments? If this doesn't work, the other losses probably won't work either.

**Comment by McHaillet** [comment]
> And another thought: since the correctness of the loss function is our biggest unknown right now, why not just disentangle the two problems and train using the misalignment std dev (which includes jitter and trajectories) for the loss calculation instead of the margin loss, then use that model to optimize alignments? If this doesn't work, the other losses probably won't work either.

Great idea! Definitely best way forward.

**Comment by McHaillet** [comment]
Done ✅
